person.rb

```diff
@@ -3,16 +3,22 @@
 require_relative 'trimmer_decorator'
 
 class Person < Nameable
+  attr_reader :id, :rentals
+  attr_accessor :age, :name
+
   def initialize(age, name = 'Unknown', parent_permission: true)
     super()
     @id = Random.rand(1..1000)
     @age = age
     @name = name
     @parent_permission = parent_permission
+    @rentals = []
   end
 
-  attr_reader :id
-  attr_accessor :age, :name
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.person = self
+  end
```

Comment:
Great job so far. Please include the parameters which are necessary for book rentals which means `date` and `book`. Logically it should and it's one of the project requirements. So remove rentals from line 18 and use `date` and `book` instead of it.

![image](https://github.com/mohammadnewton/school-library/assets/101523582/e618a8fa-79d9-420e-825f-d457db314093)
 You can visit this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for more 

2.It would be better to  `Rental.new(date, self, book)`  rather than pushing to the existing one.  Here is the difference. Your method expects an existing rental object (rentals) and tries to associate it with the book. It's not creating a new rental but rather attempting to modify an existing one. But this one ( `Rental.new(date, self, book)` ) method creates a new Rental object, associating it with the current book instance (self) and a book. It follows the logic of a "has-many" association by creating a new rental instance and not expecting an existing rental object. Kindly fix it